### PR TITLE
Add support for vector set commands

### PR DESCRIPTION
**Pull request by StefanPalashev**
# Add Support for Redis Vector Set Commands

This Pull Request implements the full suite of Redis Vector Set commands as defined in the [Redis documentation](https://redis.io/docs/latest/develop/data-types/vector-sets/), along with unit tests to verify correctness and expected behavior.

The following commands have been implemented:
- ➕ `VADD` – Add a new element to a vector set.
- 🧮 `VCARD` – Return the number of elements in a vector set.
- 📏 `VDIM` – Return the dimensionality of a vector set.
- 🧬 `VEMB` – Return the approximate vector associated with a given element in the vector set.
- 🔍 `VGETATTR` – Retrieve attribute values associated with an element of a vector set.
- ℹ️ `VINFO` – Return information about a vector set.
- 🔗 `VLINKS` – Return the neighbors of a specified element in a vector set.
- 🎲 `VRANDMEMBER` – Return one or more random elements from a vector set.
- ➖ `VREM` – Remove an element from a vector set.
- 🛠️ `VSETATTR` – Associate a JSON object with an element in a vector set.
- 🧠 `VSIM` – Return elements similar to a given vector or element. 

Command aliases:
- ❌ `VDELATTR` - A utility command that can be used to remove the associated attributes of an element in a vector set. It is an alias for `VSETATTR` with an empty string.

## 🧪 Tests

Each command is covered by unit tests that verify:

- Correct output for valid inputs
- Expected behavior for edge cases (e.g., non-existent keys or elements)
- Consistency with the Redis vector set specification

The tests serve both as validation and usage examples for future contributors.

---

### CODE REVIEW

File: redis/src/commands/mod.rs [hunk 1]
```diff
@@ -1245,6 +1248,123 @@ implement_commands! {
...
+    }
+
+    /// Associate a JSON object with an element in a vector set.
+    /// Use this command to store attributes that can be used in filtered similarity searches with VSIM.
+    /// [Redis Docs](https://redis.io/commands/VSETATTR)
+    fn vsetattr<K: ToRedisArgs, E: ToRedisArgs, J: Serialize>(key: K, element: E, json_object: &'a J) -> (bool) {
+        let attributes_json = match serde_json::to_value(json_object) {
+            Ok(serde_json::Value::String(s)) if s.is_empty() => "".to_string(),
+            _ => serde_json::to_string(json_object).unwrap(),
+        };
```
Comment (atakavci) [inline redis/src/commands/mod.rs:1304]: what happens if we drop this? it passes a "`\"\"`" ?

**Review by atakavci — COMMENTED** [review]
good job!! looks like we're just about ready for a PR.
current state is solid, but i left a couple of comments that can help improve it even further. 
let's see what we can do.

File: redis/src/commands/mod.rs (hunk 1 above)
Comment (StefanPalashev) [reply redis/src/commands/mod.rs:1304]: Yes, which is not the intended behavior. An empty string should remove the attributes.

File: redis/src/types.rs [hunk 2]
```diff
@@ -2933,6 +2946,129 @@ impl ToRedisArgs for ExpireOption {
...
+                out.write_arg(b"FP32");
+                let mut writer = out.writer_for_next_arg();
+                for &f in *vector {
+                    writer.write_all(&f.to_le_bytes()).unwrap();
+                }
+            }
+            VectorSimilaritySearchInput::Values(embeddings) => {
+                out.write_arg(b"VALUES");
+
+                match embeddings {
```
[the remark this answers is not in the document]
Comment (StefanPalashev) [reply redis/src/types.rs]: It removed the duplicated match expressions. Thanks!
